diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index a0d6df85..4152e52c 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -171,7 +171,7 @@ set(_ctk_static_suffix "")
 set(_ctk_fft_static_suffix "")
 if(CUDA_STATIC_MATH_LIBRARIES)
   set(_ctk_static_suffix "_static")
-  set(_ctk_fft_static_suffix "_static_nocallback")
+  set(_ctk_fft_static_suffix "_static")
 endif()
 
 if(NOT DISABLE_OPENMP)
